cylc/flow/task_pool.py

```diff
@@ -1181,7 +1182,11 @@ def spawn_on_output(self, itask, output, forced=False):
                 )
                 # self.workflow_db_mgr.process_queued_ops()
 
-            elif (itask.flow_nums or forced) and not itask.flow_wait:
+            elif (
```

Comment:
IMO these conditions are complex enough that a comment would be nice.

Reply:
will do ...

Reply:
Done.